### PR TITLE
Integrate flashcards

### PR DESCRIPTION
**Pull request by myluki2000**
## Description of changes
Integrates the flashcard service into the gateway. Also includes some minor changes to user authentication. User data is now sent to the services as JSON in the "CurrentUser" http header.
  
## How has this been tested?
Please describe the test strategy you followed.

- [ ] automated unit test
- [ ] automated integration test
- [ ] automated acceptance test
- [x] manual, exploratory test

Manually tested using GraphiQL that flashcard service queries & mutations are accessible and that flashcard sets can be accessed via the FlashcardSetAssessment per the unified schema.
    
## Checklist before requesting a review
- [x] My code follows the [coding guidelines](https://github.com/IT-REX-Platform/wiki/blob/main/dev-manuals/backend/coding-guidelines.md) of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My code fulfills all acceptance criteria
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] The test coverage (line and branch) is reasonably high, especially on Service classes and other classes with complex logic
- [ ] I have made corresponding changes to the documentation
- [ ] I have added explanation of architectural decision and rationales to [wiki/adr](https://github.com/IT-REX-Platform/wiki/tree/main/adr)
- [ ] I have updated the changes in the ticket description

## Checklist for reviewer
- [ ] The code works and does not throw errors
- [ ] The code is easy to understand and there are no confusing parts
- [ ] The code follows the [coding guidelines](https://github.com/IT-REX-Platform/wiki/blob/main/dev-manuals/backend/coding-guidelines.md) of this project
- [ ] The code change accomplishes what it is supposed to do
- [ ] I cannot think of any use case in which the code does not behave as intended
- [ ] The added and existing tests reasonably cover the code change
- [ ] I cannot think of any test cases, input or edge cases that should be tested in addition
- [ ] Description of the change is included in the documentation

### CODE REVIEW

File: .meshrc.yaml [hunk 1]
```diff
@@ -14,6 +14,12 @@ sources:
     handler:
       graphql:
         endpoint: "{env.MEDIA_SERVICE_URL}"
+        operationHeaders:
+          CurrentUser: "{context.currentUserJson}"
```
Comment (PaulBredl) [inline .meshrc.yaml:18]: Ich finds irgendwie weird, dass als header feld ein Json string übergeben wird, aus dem man die user id wieder rausziehen muss. Kann man nicht die Header felder einzeln setzen, bzw zumindest die UserID? der Rest wird meistens eh nicht gebraucht

Comment (myluki2000) [reply .meshrc.yaml:18]: Später werden da ja auch noch rollen und kurse auf die man zugriff hat reinkommen. Fand das so am elegantesten weil man dann nur das Model anpassen muss im Gateway und im gits-common, anstatt einzeln die header von Hand auslesen zu müssen.

**Review by PaulBredl — APPROVED** [review]
